### PR TITLE
Add export to CSV option to Discover page

### PR DESCRIPTION
**Pull request by pdcp1**
This closes https://github.com/responsible-ai-collaborative/aiid/issues/1235

image

### CODE REVIEW

File: site/gatsby-site/src/components/discover/Controls.js [hunk 1]
```diff
@@ -36,6 +37,10 @@ const Controls = () => {
         </div>
 
         <div className="flex">
+          <div className="bootstrap place-self-center">
```
Comment (cesarvarela) [inline site/gatsby-site/src/components/discover/Controls.js:40]: I think we should stop using `bootstrap` stuff?

**Review by cesarvarela — COMMENTED** [review]
This only exports the first page. Is that what's expected?

File: site/gatsby-site/src/components/discover/Controls.js (hunk 1 above)
Comment (pdcp1) [reply site/gatsby-site/src/components/discover/Controls.js]: Changed to flowbite button

**Comment by pdcp1** [comment]
> This only exports the first page. Is that what's expected?

That's what the Algolia React component support. Only the visible page results are accessible to export.
To export all results we need to change the export component to get the results in another way, maybe a direct Algolia API call.
What do you think @LKchemposer? Is it OK to only export the visible page results? You can play with it [here](https://deploy-preview-1629--staging-aiid.netlify.app/apps/discover/?display=details&is_incident_report=true&page=1).

cc: @smcgregor

**Comment by cesarvarela** [comment]
I'll merge after we get a response to the questions above.

**Comment by LKchemposer** [comment]
@pdcp1 @cesarvarela This looks great! I think this is OK to roll out for now as a demo, although the feature might benefit from a few things from the UX perspective, in order of importance:

1. If only results of the first page is exported, there needs to be some communication about that, maybe via a notification?
2. The fields in the CSV does not contain `incident_id` (and `incident_title`) which are most likely what people want to have when exporting
3. Naming the button `Export` with the icon is friendly to a wider range of users
4. Not sure if the `text` field is needed in the CSV, as it takes a large chunk of the data. my guess is that people who would use the export feature do so to keep the db's indexes of the reports, rather than the text of the reports themselves. for the text, we can direct them to the `Download Complete Database` link at the bottom of the site instead, maybe also via the notification?

Please let me know your thoughts on these. I'd imagine some of these are easier to implement than others.

**Comment by pdcp1** [comment]
@LKchemposer Thanks for your feedback.
This is how it looks like after the changes:

Screenshot 2023-02-23 at 17 25 53

Note: We couldn't add the `incident title` field to the result set since it's not an Algolia field.

**Comment by pdcp1** [comment]
@cesarvarela the build passed after a second run here https://app.netlify.com/sites/staging-aiid/deploys/63f7da6a25b0bc1846698f04

**Review by cesarvarela — COMMENTED** [review]
The test failing is passing on staging with no problem. I think there is something in this changeset that's creating some flakiness 🤔

**Comment by cesarvarela** [comment]
Scratch that, it is failing here too: https://github.com/responsible-ai-collaborative/aiid/pull/1651, I'll take a look at why it is flaky

**Comment by pdcp1** [comment]
@cesarvarela the tests pass now

**Pull request merged**